src/main/java/org/medicmobile/webapp/mobile/MedicAndroidJavascript.java

```diff
@@ -230,6 +254,96 @@ public void sms_send(String id, String destination, String content) throws Excep
 		}
 	}
 
+	@org.xwalk.core.JavascriptInterface
+	@android.webkit.JavascriptInterface
+	public String getDeviceInfo() {
+		try {
+			if (activityManager == null) {
+				return jsonError("ActivityManager not set. Cannot retrieve RAM info.");
+			}
+
+			if (connectivityManager == null) {
+				return jsonError("ConnectivityManager not set. Cannot retrieve network info.");
+			}
+
+			String versionName = parent.getPackageManager()
+					.getPackageInfo(parent.getPackageName(), 0)
+					.versionName;
+			JSONObject appObject = new JSONObject();
+			appObject.put("version", versionName);
+
+			String androidVersion = Build.VERSION.RELEASE;
+			int osApiLevel = Build.VERSION.SDK_INT;
+			String osVersion = System.getProperty("os.version") + "(" + android.os.Build.VERSION.INCREMENTAL + ")";
+			JSONObject softwareObject = new JSONObject();
+			softwareObject
+					.put("androidVersion", androidVersion)
+					.put("osApiLevel", osApiLevel)
+					.put("osVersion", osVersion);
+
+			String device = Build.DEVICE;
+			String model = Build.MODEL;
+			String manufacturer = Build.BRAND;
+			String hardware = Build.HARDWARE;
+			Map<String, String> cpuInfo = getCPUInfo();
+			JSONObject hardwareObject = new JSONObject();
+			hardwareObject
+					.put("device", device)
+					.put("model", model)
+					.put("manufacturer", manufacturer)
+					.put("hardware", hardware)
+					.put("cpuInfo", new JSONObject(cpuInfo));
+
+			File dataDirectory = Environment.getDataDirectory();
+			StatFs dataDirectoryStat = new StatFs(dataDirectory.getPath());
+			long dataDirectoryBlockSize = dataDirectoryStat.getBlockSizeLong();
+			long dataDirectoryAvailableBlocks = dataDirectoryStat.getAvailableBlocksLong();
+			long dataDirectoryTotalBlocks = dataDirectoryStat.getBlockCountLong();
+			long freeMemorySize = dataDirectoryAvailableBlocks * dataDirectoryBlockSize;
+			long totalMemorySize = dataDirectoryTotalBlocks * dataDirectoryBlockSize;
+			JSONObject memoryObject = new JSONObject();
+			memoryObject
+					.put("free", freeMemorySize)
+					.put("total", totalMemorySize);
+
+			MemoryInfo memoryInfo = new ActivityManager.MemoryInfo();
+			activityManager.getMemoryInfo(memoryInfo);
+			long totalRAMSize = memoryInfo.totalMem;
+			long freeRAMSize = memoryInfo.availMem;
+			long thresholdRAM = memoryInfo.threshold;
+			Runtime runtime = Runtime.getRuntime();
```

Comment:
Sorry I missed when this was suggested so I didn't comment on it.

This may actually be more of a question for @kennsippell since he suggested it.

Because for us the JVM does basically nothing (because most of the work is XWalk -> Chromium which doesn't run in the JVM), does this value and other values in this class indicate useful things for us? If so why?

If it *is* useful, I wonder then if we want to pull more from it. Things that look interesting to me: `availableProcessors`, `freeMemory`, `maxMemory` (the one we have), `totalMemory`.